src/transformers/quantizers/auto.py

```diff
@@ -0,0 +1,144 @@
+# Copyright 2024 The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import warnings
+from typing import Dict, Optional, Union
+
+from ..models.auto.configuration_auto import AutoConfig
+from ..utils.quantization_config import (
+    AwqConfig,
+    BitsAndBytesConfig,
+    GPTQConfig,
+    QuantizationConfigMixin,
+    QuantizationMethod,
+)
+from .quantizer_awq import AwqQuantizer
+from .quantizer_bnb import Bnb4BitHFQuantizer, Bnb8BitHFQuantizer
+from .quantizer_gptq import GptqHFQuantizer
+
+
+AUTO_QUANTIZER_MAPPING = {
+    "awq": AwqQuantizer,
+    "bitsandbytes_4bit": Bnb4BitHFQuantizer,
+    "bitsandbytes_8bit": Bnb8BitHFQuantizer,
+    "gptq": GptqHFQuantizer,
+}
+
+AUTO_QUANTIZATION_CONFIG_MAPPING = {
+    "awq": AwqConfig,
+    "bitsandbytes_4bit": BitsAndBytesConfig,
+    "bitsandbytes_8bit": BitsAndBytesConfig,
+    "gptq": GPTQConfig,
+}
+
+
+class AutoQuantizationConfig:
+    """
+    The Auto-HF quantization config class that takes care of automatically dispatching to the correct
+    quantization config given a quantization config stored in a dictionary.
+    """
+
```

Comment:
```suggestion
    quant_method = None
```

Reply:
This is not needed since the quantization config do not inherit from that class